sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -448,42 +572,106 @@ def get_test_params(cls, parameter_set="default"):
                     "deterministic": True,
                 }
             )
-
+        params_no_broadcasting = [
+            dict(p, **{"broadcasting": False}) for p in test_params
+        ]
+        test_params.extend(params_no_broadcasting)
         return test_params
 
 
 class PyTorchDataset(Dataset):
     """Dataset for use in sktime deep learning forecasters."""
 
-    def __init__(self, y, seq_len, fh=None, X=None):
-        self.y = y.values
-        self.X = X.values if X is not None else X
-        self.seq_len = seq_len
+    def __init__(
+        self,
+        y: pd.DataFrame,
+        seq_len: int,
+        fh=None,
+        X: pd.DataFrame = None,
+        batch_size=8,
+        no_size1_batch=True,
+    ):
+        if not isinstance(y.index, pd.MultiIndex):
+            self.y = np.array(y.values, dtype=np.float32).reshape(1, len(y), 1)
+            self.X = (
+                np.array(X.values, dtype=np.float32).reshape(
+                    (1, len(X), len(X.columns))
+                )
+                if X is not None
+                else X
+            )
+        else:
+            self.y = _frame2numpy(y)
+            self.X = _frame2numpy(X) if X is not None else X
+
+        self._num, self._len, _ = self.y.shape
         self.fh = fh
+        self.seq_len = seq_len
+        self.batch_size = batch_size
+        self.no_size1_batch = no_size1_batch
 
     def __len__(self):
         """Return length of dataset."""
-        return max(len(self.y) - self.seq_len - self.fh + 1, 0)
+        true_length = self._num * max(self._len - self.seq_len - self.fh + 1, 0)
+        if self.no_size1_batch and true_length % self.batch_size == 1:
+            return true_length - 1
+        else:
+            return true_length
 
     def __getitem__(self, i):
         """Return data point."""
-        from torch import from_numpy, tensor
-
-        hist_y = tensor(self.y[i : i + self.seq_len]).float()
+        from torch import tensor
+
+        m = i % (self._len - self.seq_len - self.fh + 1)
+        n = int((i - m) / self._len)
```

Comment:
```suggestion
        n = (i - m) // self._len
```
as an alternative to the cast.

Reply:
Actully you found a bug! It should be `int((i - m) / (self._len - self.seq_len - self.fh + 1))`. I will change it to 
```py
m = i % self._len_single
n = i // self._len_single
```

where `self._len_single = self._len - self.seq_len - self.fh + 1`